core/domain/classifier_domain_test.py

```diff
@@ -269,3 +269,59 @@ def test_validation(self):
         with self.assertRaisesRegexp(utils.ValidationError, (
             'Expected training_data to be a list')):
             training_job.validate()
+
+
+class ClassifierExplorationMappingDomainTests(test_utils.GenericTestBase):
+    """Tests for the ClassifierExplorationMapping domain."""
+
+    def _get_mapping_from_dict(self, mapping_dict):
+        mapping = classifier_domain.ClassifierExplorationMapping(
+            mapping_dict['exp_id'],
+            mapping_dict['exp_version'],
+            mapping_dict['state_name'],
+            mapping_dict['classifier_id'])
+
+        return mapping
+
+    def test_to_dict(self):
+        expected_mapping_dict = {
+            'exp_id': 'exp_id1',
+            'exp_version': 2,
+            'state_name': 'state_name1',
+            'classifier_id': 'classifier_id1'
+        }
+        observed_mapping = self._get_mapping_from_dict(
+            expected_mapping_dict)
+        self.assertDictEqual(expected_mapping_dict,
+                             observed_mapping.to_dict())
+
+    def test_validation(self):
```

Comment:
@seanlip . Ah, maybe all strings get converted to unicode in the storage layer for StringProperty? Like we never really bothered converting all StringProperty attributes to unicode before we stored it, did we? That should be something internal then right? Maybe the StringProperty is of type unicode, but we require utf-8 strings in the logic layer. So, I sort of don't see an issue in keeping all state_names as utf-8 strings. It shouldn't really be much different from using exp_id as a string right? 

Phew. This is sort of confusing me here. But, fundamentally, I'm still not sure how we will have an issue if we keep the ID as utf-8 strings. utf-8 is a pretty good encoding that can handle unicode strings like they were any python string (printing, concatenating and the like). If I'm wrong somewhere in this, please correct me.